### PR TITLE
WIP - generate pull request validation report

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

**Comment by jrudolph** [comment]
Btw. I added a few errors into here by purpose to see what the output will be...

**Comment by ktoso** [comment]
Mmm, would love that at some point 😉

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by jrudolph** [comment]
I fixed the Jenkins plugin so that we could use the "comment file" feature where the build can generate a file that is picked up and added to the github commit.

https://github.com/jenkinsci/ghprb-plugin/pull/512

As can be seen above, the reporting itself still needs some improvements.

**Comment by raboof** [comment]
Cool idea! Looks like this uses some things that became 'internal' in sbt 1 (such as `Aggregation`), so it might take some work to rebase...

**Comment by jrudolph** [comment]
Resolved sbt 1 issues. It seems to be somewhat working again. The ghprb PR has been merged but no new version was released, so I might try to install a locally built version of it again to see it in action.

**Comment by jrudolph** [comment]
PLS BUILD

**Comment by raboof** [comment]
This is starting to look really nice!

File: akka-http-marshallers-scala/akka-http-spray-json/src/test/scala/akka/http/scaladsl/marshallers/sprayjson/SprayJsonSupportSpec.scala [hunk 1]
```diff
@@ -33,7 +33,7 @@ class SprayJsonSupportSpec extends WordSpec with Matchers with ScalaFutures {
       val js = Marshal(init).to[MessageEntity].futureValue
       val example = Unmarshal(js).to[Example].futureValue
 
-      example should ===(init)
+      example should ===(null) //===(init)
```
Comment (johanandren) [inline akka-http-marshallers-scala/akka-http-spray-json/src/test/scala/akka/http/scaladsl/marshallers/sprayjson/SprayJsonSupportSpec.scala]: We don't want to merge these example failures, right?

**Comment by raboof** [comment]
(we might still have to configure jenkins to read the 'commit file', let's see though)

**Pull request merged**